### PR TITLE
feat: add `deployment_id` to the ui and licenses

### PR DESCRIPTION
**Pull request by kylecarbs**
Allows customers to send us the deployment ID easily. I put this in the user dropdown because it's so global. If users find this confusing, we can happily move it!
image

Soon, licenses can be generated with specific deployment IDs to match against. This is to prevent customers from having a bunch of Coder deployments with the same license.

### CODE REVIEW

**Review by ericpaulsen — APPROVED** [review]
i'll let @johnstcn reviewing the go code - but the UI LGTM.

File: site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx [hunk 1]
```diff
@@ -161,15 +163,51 @@ export const UserDropdownContent: FC<UserDropdownContentProps> = ({
...
+            css={[styles.footerText, styles.buildInfo]}
+            href={buildInfo?.external_url}
+            target="_blank"
+            rel="noreferrer"
+          >
+            {buildInfo?.version} <LaunchIcon />
+          </a>
+        </Tooltip>
+
+        {buildInfo?.deployment_id && (
```
Comment (Parkreiner) [inline site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx:178]: Is there a chance that `deployment_id` could ever be an empty string?

I'm wondering if the `buildInfo?.deployment_id &&` could be swapped for `Boolean(buildInfo?.deployment_id) &&` to be a bit more explicit about disallowing any kind of falsey value

File: site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx [hunk 2]
```diff
@@ -161,15 +163,51 @@ export const UserDropdownContent: FC<UserDropdownContentProps> = ({
...
+              buttonStyles={css`
+                width: 16px;
+                height: 16px;
+
+                svg {
+                  width: 16px;
+                  height: 16px;
+                }
+              `}
+            />{" "}
```
Comment (Parkreiner) [inline site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx:208]: There's a stay whitespace character here that Prettier didn't get rid of

File: site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx (hunk 1 above)
Comment (kylecarbs) [reply site/src/modules/dashboard/Navbar/UserDropdown/UserDropdownContent.tsx:178]: Agreed!

File: .github/workflows/mlc_config.json [hunk 3]
```diff
@@ -17,6 +17,9 @@
     },
     {
       "pattern": "tailscale.com"
+    },
+    {
+      "pattern": "wireguard.com"
```
Comment (matifali) [inline .github/workflows/mlc_config.json:22]: sweet <3

**Comment by matifali** [comment]
@kylecarbs it is not showing up on dev.coder.com although the version [v2.10.0-devel+1bda8a085](https://github.com/coder/coder/commit/1bda8a085687a4176bf5aa0a2cea718e1310af6c) have been deployed successfully.

**Comment by kylecarbs** [comment]
Hmm, I'll check why that is. Odd

**Comment by kylecarbs** [comment]
@matifali fixed!

**Pull request merged**